CMakeLists.txt

```diff
@@ -284,6 +284,22 @@ elseif(MSVC)
   add_definitions("/MP")
 endif()
 
+# Append or replace CMAKE_CXX_FLAGS.
+if(DEFINED FLATBUFFERS_CXX_FLAGS AND NOT EXISTS "${CMAKE_TOOLCHAIN_FILE}")
+  string(FIND "${FLATBUFFERS_CXX_FLAGS}" "#" pos)
+  if(${pos} EQUAL 0)
+    # Replace CMAKE_CXX_FLAGS by FLATBUFFERS_CXX_FLAGS if it starts with '#' symbol.
```

Comment:
whats an example of this usage, and why is it needed? All this string manipulation scares me :)

Reply:
The `#` option is a foresight. It can be useful for debugging and prototyping.
But CMakeLists.txt already has "${CMAKE_TOOLCHAIN_FILE}" option. The new `#` duplicates it so better to remove it.